src/AzureAppConfigurationImpl.ts

```diff
@@ -458,6 +443,9 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
 
         // try refresh if any of watched settings is changed.
         let needRefresh = false;
+        if (this.#watchAll) {
+            needRefresh = await this.#checkKeyValueCollectionChanged(this.#keyValueSelectors);
+        }
         for (const sentinel of this.#sentinels.values()) {

```

Comment:
What's the response when sentinels not existed in app config store?

Reply:
The sdk will throw REST exception with status code 404. However, the exception will be caught by the provider [here](https://github.com/Azure/AppConfiguration-JavaScriptProvider/blob/main/src/AzureAppConfigurationImpl.ts#L527)